### DISCUSSION TITLE
Require SMS OTP after Email + Password login

### DISCUSSION THREAD

**Discussion by Jammy108** [supabase/supabase#26978] [Questions]
Hi,

I'm considering implementing a flow where users authenticate with an email and password, followed by receiving an OTP via email or phone for verification. This seems more secure than passwordless authentication. However, I understand that Supabase MFA currently only supports TOTP. Is there a way to do this that I'm missing?

Would it be possible to achieve this with the new password verification auth hook? My plan is to deny all password logins through the hook, but send an OTP if the password is valid. The user would then enter this code, and I'd authenticate them using "signInWithOtp."

Is this a stupid idea? The application handles fairly sensitive data, so I want a secure method without relying on authenticator apps, if possible. Thanks!

**Comment by abhishek-yelley-sails** [comment]
Hey @Jammy108! I have been trying to do the same. Email+Password login and then an OTP verification via email.
For your particular case, supabase implemented a [phone based MFA method](https://supabase.com/docs/guides/auth/auth-mfa/phone).

My project requires for this to be done via email and I took your idea and tried implementing it like so: https://www.reddit.com/r/Supabase/comments/1eyysjt/comment/ljnxgfd/

I am not sure if this is secure/good. Would love to hear how you did it and also if you have any suggestions or ideas.
